src/CarbonAware.WebApi/test/unitTests/CarbonAwareControllerTests.cs

```diff
@@ -131,6 +132,56 @@ public async Task GetAverageCarbonIntensity_SuccessfulCallReturnsOk(string locat
         Assert.AreEqual(expectedContent, actualContent);
     }
 
+    /// <summary>
+    /// Tests that successfull call the average carbon intensity for a batch of requests on valid input
+    /// </summary>
+    [Test]
+    public async Task CalculateAverageCarbonIntensityBatch_ValidInput()
+    {
+        // Arrange
+        string location = "Sydney";
+        var start1 = new DateTimeOffset(2022, 3, 1, 0, 0, 0, TimeSpan.Zero);
+        var end1 = new DateTimeOffset(2022, 3, 1, 1, 0, 0, TimeSpan.Zero);
+
+        var start2 = new DateTimeOffset(2022, 3, 2, 0, 0, 0, TimeSpan.Zero);
+        var end2 = new DateTimeOffset(2022, 3, 2, 1, 0, 0, TimeSpan.Zero);
+        double data = 0.7;
+        var controller = new CarbonAwareController(this.MockCarbonAwareLogger.Object, CreateCarbonAwareAggregatorWithAverageCI(data).Object);
+
+        var request1 = new CarbonIntensityBatchDTO { Location = location, StartTime = start1, EndTime = end1 };
+        var request2 = new CarbonIntensityBatchDTO { Location = location, StartTime = start2, EndTime = end2 };
+        var requestList = new List<CarbonIntensityBatchDTO> { request1, request2 };
+        // Act
+        var actualContent = new List<CarbonIntensityDTO> { };
+        await foreach (var item in controller.GetAverageCarbonIntensityBatch(requestList))
+        {
+            actualContent.Add(item);
+        }
+
+        // Assert
+        var expectedContent1 = new CarbonIntensityDTO { Location = location, StartTime = start1, EndTime = end1, CarbonIntensity = data };
+        var expectedContent2 = new CarbonIntensityDTO { Location = location, StartTime = start2, EndTime = end2, CarbonIntensity = data };
+        var expectedContent = new List<CarbonIntensityDTO> { expectedContent1, expectedContent2 };
+        CollectionAssert.AreEqual(expectedContent, actualContent);
+    }
+
+    /// <summary>
+    /// Tests that are missing a location and thus throw an ArgumentException error
+    /// </summary>
+    [TestCase(null, "2022-03-07T01:00:00", "2022-03-07T01:00:00", TestName = "CalculateAverageCarbonIntensityBatch throws exception for no location")]
```

Comment:
nit: Is this the only test? if that is the case, maybe no need to do the `[TestCase]...` and perform just this specific one "internally".

Reply:
following route that juzuluag suggested, since is only one case, and similar to existing _BatchForecast_NoLocations_ThrowsException_